lib/twitter_cldr/shared/locale.rb

```diff
@@ -0,0 +1,265 @@
+        end
+
+        protected
+
+        def normalize(locale_text)
+          Locale.new(nil).tap do |locale|
+            subtags = identify_subtags(locale_text)
+
+            until subtags.empty?
+              subtag, identities = subtags.shift
+              next if identities.empty?
+
+              identities.each do |identity|
+                unless subtag_set?(locale, identity)
+                  set_subtag(locale, identity, subtag)
+                  break
+                end
+              end
+            end
+          end
+        end
+
+        def subtag_set?(locale, identity)
+          case identity
+            when :variant
+              false
+            else
+              !!locale.send(identity)
+          end
+        end
+
+        def set_subtag(locale, identity, subtag)
+          case identity
+            when :variant
+              locale.variants << normalize_subtag(subtag, identity)
+            else
+              locale.send(
+                :"#{identity}=", normalize_subtag(subtag, identity)
+              )
+          end
+        end
+
+        def identify_subtags(locale_text)
+          split(locale_text).map do |subtag|
+            identities = identify_subtag(subtag)
+            [subtag, identities]
+          end
+        end
+
+        def identify_subtag(subtag)
+          [].tap do |types|
+            types << :language if language?(subtag)
+            types << :script   if script?(subtag)
+            types << :region   if region?(subtag)
+            types << :variant  if variant?(subtag)
+          end
+        end
+
+        def language?(subtag)
+          subtag = normalize_subtag(subtag, :language)
+          languages.include?(subtag) ||
+            aliases_resource[:language].include?(subtag.to_sym)
+        end
+
+        def script?(subtag)
+          subtag = normalize_subtag(subtag, :script)
+          scripts.include?(subtag)
+        end
+
+        def region?(subtag)
+          subtag = normalize_subtag(subtag, :region)
+          territories.include?(subtag) ||
+            aliases_resource[:territory].include?(subtag.to_sym)
+        end
+
+        def variant?(subtag)
+          subtag = normalize_subtag(subtag, :variant)
+          variants.include?(subtag)
+        end
+
+        def normalize_subtag(subtag, identity)
+          case identity
+            when :language
+              subtag.downcase
+            when :script
+              subtag.capitalize
+            when :region, :variant
+              subtag.upcase
+          end
+        end
+
+        def replace_deprecated_subtags(locale)
+          replace_deprecated_language_subtags(locale)
+          replace_deprecated_territory_subtags(locale)
+        end
+
+        def replace_deprecated_language_subtags(locale)
+          language = locale.language ? locale.language.to_sym : nil
+          if found_alias = aliases_resource[:language][language]
+            replacement = if found_alias.is_a?(Array)
```

Comment:
When would this be an array? I couldn't find any arrays in the `:language` section of `aliases.yml` file.